store/db.go

```diff
@@ -49,15 +51,17 @@ func newDBLock(dirPath string) (*dbLock, error) {
 }
 
 func (dl *dbLock) lock() error {
+	dl.Lock()
 	if err := dl.fl.ExclusiveLock(); err != nil {
+		dl.Unlock()
 		return err
 	}
-	dl.Lock()
 	return nil
 }
 
 func (dl *dbLock) unlock() error {
 	if err := dl.fl.Unlock(); err != nil {
+		dl.Unlock()
```

Comment:
> If the fl.Unlock here fails

Looking at `man 2 flock` looks like the only way an unlock will fail is if the underlying fd has been closed. This should be a programming error. But I'm not sure if there'll be other cases.


How to handle this possibility?
usually rkt commands will get the error and exit, instead the rkt api service will continue to use that store instance. Probably, it should handle this error specially, closing that store instance and creating a new one (this needs to be correctly handled in different ways)